packages/client3/hooks/application-hooks.ts

```diff
@@ -0,0 +1,13 @@
+import { useContext } from "react";
+
+import { ApplicationContext } from "@/components/layout";
+
+export function useApplicationState() {
+  const { isState1Selected } = useContext(ApplicationContext);
```

Comment:
What is the state1 tho? Either document or change the name of the variable to something more readable.

Reply:
```
  states: {
    isState1Selected: boolean;
    state1: string;
    state2: string;
  };
``` 

These come from the toggle props? As recommended by you and Vineeth when the Toggle component was being reviewed.